base/src/main/java/proguard/ProGuard.java

```diff
@@ -195,6 +196,13 @@ public void execute() throws Exception
                 new ListParser(new NameParser()).parse(configuration.optimizations) :
                 new ConstantMatcher(true);
 
+            // Inline kotlin lambdas passed as arguments to methods as long as they are
+            // created in the arguments or created in a local variable and passed as an argument
+            if (configuration.lambdaInlining)
+            {
+                inlineLambdas();
```

Comment:
this should probably happen inside the optimize method instead, and run once per optimization pass